lib/internal/encoding.js

```diff
@@ -411,15 +411,7 @@ function makeTextDecoderICU() {
 
     decode(input = empty, options = kEmptyObject) {
       validateDecoder(this);
-      if (isAnyArrayBuffer(input)) {
-        try {
-          input = lazyBuffer().from(input);
-        } catch {
-          // If the buffer is detached,
-          // use an empty Uint8Array to avoid TypeError
-          input = empty;
```

Comment:
Is there a test covering this condition?

Reply:
According the pull request that added this line, there is: https://github.com/nodejs/node/commit/86088ab78e8c3739f60627ddf21f65da7796450e